### DISCUSSION TITLE
Add create listener for TracingContext and IgnoreTracingContext.

### DISCUSSION THREAD

**Discussion by weixiang1862** [apache/skywalking#11245] [Q&A] [answered]
In my company, we are use skywalking agent to develop  gray-scale publishing plugin. We use `SkyWalking Cross Process Correlation Headers Protocol` to propagate gray flag (Thanks for developing this awesome tools).

In our scenario, we need to watch the creation of `CorrelationContext` and then add some tags to it. Now, we changed agent-core codes to implement it, it is not elegant. 

If skywalking official can provider a method to listen TracingContext creation, we can implement our logic totally in skywalking plugin mechanism.

I did not found other plugins has this scenario, it is just our's. So If you feel bad with this idea, just feel free to reject it, we can try to resolve this in other ways.

![848e77b087a776ad3dc6db9d3d84b72](https://github.com/apache/skywalking/assets/16346644/6736fc3a-b4f9-45cc-8d5c-b0abfe73d67e)

**Comment by wu-sheng** [comment, marked as answer]
We can't guarantee a listener is a proper way. Your side of cases works great for yourself, that is good. But honestly, that is because your classloader structure is super easy. Most likely, you are running a simple Spring boot application only. 
Plugins' codes should not ever touch your business codes. So a listener is not a good idea. 

Watching how the kernel is running should not be a recommended way from an official perspective.

Of course,  you could build as you like, but it can't be upstream.

**Reply by weixiang1862** [reply]
Ok, thanks again for you and community, it helps us a lot.
